### DISCUSSION TITLE
Share your questions!

### DISCUSSION THREAD

**Discussion by hasherezade** [hasherezade/pe-sieve#80] [Q&A]
Do you have any question about PE-sieve, that was not yet included in [the FAQ](https://github.com/hasherezade/pe-sieve/wiki/1.-FAQ)? Feel free to ask it here...

**Comment by JemoGlemo** [comment]
I'm investigating a case where a loader modifies a single byte in the memory of the original EXE (e.g., demo.exe) after launch. I confirmed that the loader successfully writes the byte at a specific VA (verified via manual inspection), but when I run PE-sieve (32-bit) on the process, it reports no suspicious modules or patched memory.

I tried: pe-sieve.exe /pid 22564 /dump_mode 1 /shellc /threads

My question:

Does PE-sieve detect single-byte memory modifications by default?
Are there specific flags or configurations I should enable to force more sensitive patch detection?
Could alignment or section boundary issues cause it to miss very small changes?

Any guidance or tuning recommendations would be appreciated.

**Reply by hasherezade** [reply]
hi @JemoGlemo ! When PE-sieve scans the executable loaded in memory, it focuses only on the areas containing the code. Was the patch that you observed located within the code section? If it was somewhere else, in non-executable section, it is normal that it haven't been caught. Can you provide some more details, screenshots, or ideally, the sample itself, and the precise patch location, so that I could reproduce it on my side?

**Reply by JemoGlemo** [reply]
Hello @hasherezade 

Certainly. I've attached both the main executable file and its corresponding loader. The scenario is straightforward: I used the loader to modify a value at virtual address 0x403000, changing it from 52 (the letter 'R') to 47 (the letter 'G'). The loader was created using an old tool called (PatchCreationWizard 😄) to patch the memory at runtime.

In summary, what I need is to identify which bytes are being written during the execution of the loader.

[REGKEY.zip](https://github.com/user-attachments/files/20892498/REGKEY.zip)

**Reply by hasherezade** [reply]
Thank you. I checked it and as I thought - the answer to your question is simple. The modification that you make is in non-executable area. It is in the `.data` section of the PE.

![sec1](https://github.com/user-attachments/assets/ae7fdf65-32df-4c60-a73f-ccae95dfc001)

![sec2](https://github.com/user-attachments/assets/b8743330-32dc-448d-86ea-70fd3d767326)

![sec3](https://github.com/user-attachments/assets/58acfee1-89b1-48e9-9bd4-e9b4f168a2d7)
(RW - Read Write permissions - not executable)

So, PE-sieve works as intended.
It is not about the size of the patch, but where it is located. If it was in the executable are, it would be detected.

**Reply by JemoGlemo** [reply]
Thank you for the clarification! That makes perfect sense now. I’ve now applied the patch to the .text (code) section instead, and as expected, PE-sieve successfully detected it. Everything is working as intended 😄
